l1-contracts/src/core/messagebridge/frontier_tree/Frontier.sol

```diff
@@ -16,33 +16,34 @@ contract FrontierMerkle is IFrontier {
   // for the zeros at each level. This would save gas on computations
   mapping(uint256 level => bytes32 zero) public zeros;
 
-  constructor(uint256 _depth) {
-    DEPTH = _depth;
-    SIZE = 2 ** _depth;
+  constructor(uint256 _height) {
+    HEIGHT = _height;
+    SIZE = 2 ** _height;
 
     zeros[0] = bytes32(0);
-    for (uint256 i = 1; i <= DEPTH; i++) {
+    for (uint256 i = 1; i <= HEIGHT; i++) {
       zeros[i] = sha256(bytes.concat(zeros[i - 1], zeros[i - 1]));
     }
   }
 
-  function insertLeaf(bytes32 _leaf) external override(IFrontier) {
+  function insertLeaf(bytes32 _leaf) external override(IFrontier) returns (uint256) {
     uint256 level = _computeLevel(nextIndex);
     bytes32 right = _leaf;
     for (uint256 i = 0; i < level; i++) {
       right = sha256(bytes.concat(frontier[i], right));
     }
     frontier[level] = right;
-    nextIndex++;
+
+    return nextIndex++;
```

Comment:
This looks kinda cursed, ngl. You could just cache the value at the start (the one that is passed into `_computeLevel` and then return that at the end 🤷